### PR TITLE
maintenance[message-relayer]: make getSentMessages function more testable

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Pulls the `getSentMessages` function out of the `MessageRelayer` and adds some unit tests. Also adds some additional validation checks to each function to improve our error observability.

### CODE REVIEW

File: packages/message-relayer/src/relay-tools.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  try {
+    // Events are emitted on L2 here:
+    // https://github.com/ethereum-optimism/optimism/blob/2d352de8f257db2679b8ac6f953c146a8e99f05d/packages/contracts/contracts/optimistic-ethereum/OVM/bridge/messaging/Abs_BaseCrossDomainMessenger.sol#L88
+    // Encoding is performed here:
+    // https://github.com/ethereum-optimism/optimism/blob/2d352de8f257db2679b8ac6f953c146a8e99f05d/packages/contracts/contracts/optimistic-ethereum/OVM/bridge/messaging/Abs_BaseCrossDomainMessenger.sol#L116-L122
+    decoded = getContractInterface(
+      'OVM_L2CrossDomainMessenger'
+    ).decodeFunctionData('relayMessage', message) as any
+  } catch (err) {
+    throw new Error('unable to parse SentMessage event')
```
Comment (tynes) [inline packages/message-relayer/src/relay-tools.ts:45]: Can we include the txhash in this error message?

**Review by tynes — CHANGES_REQUESTED** [review]
Can we include the txhash in the error message?

File: packages/message-relayer/src/relay-tools.ts (hunk 1 above)
Comment (smartcontracts) [reply packages/message-relayer/src/relay-tools.ts]: Done.

**Comment by tynes** [comment]
@smartcontracts What is the status of this PR? I notice that its no longer ready for review

**Comment by smartcontracts** [comment]
> @smartcontracts What is the status of this PR? I notice that its no longer ready for review

I'm gonna close this PR since it's largely superseded by #952

**Pull request closed without merging**